### DISCUSSION TITLE
EKS vs ECS for self-hostes GitHub runners ?

### DISCUSSION THREAD

**Discussion by przemolb** [community/community#59917] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello,

we are going to deploy self-hosted GitHub runners  sooner or later. Initially we will have a few 10s runners but at some point we might get to 1000s. We decided to run them on either ECS (simpler) or EKS (we have enough internal knowledge).
What about scalability of ECS vs EKS ? DOes ECS or EKS is better than the other ? If so then why ?

Thanks

**Comment by mateothegreat** [comment]
If you want granular control I'd go with EKS and ARC (Actions Runner Controller) to manage the lifecycle of the runners (https://github.com/actions/actions-runner-controller).

I have a pretty decent setup that has worked for a couple of years now combining auto-scaling and metrics to instrument a "scaling-from-zero" strategy to save on costs while still providing low enough latency to the development lifecycle that I couldn't live without it.

If you must use ECS you're going to have to more than likely roll your own solution that has already been implemented with ARC.

TL;DR: EKS + ARC!

**Reply by przemolb** [reply]
What exactly has to be implemented manually while using ECS ?
